extensions/logging-gelf/runtime/src/main/java/io/quarkus/logging/gelf/GelfConfig.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.logging.gelf;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.RUN_TIME, name = "log.handler.gelf")
+public class GelfConfig {
+    /**
+     * Determine whether to enable the GELF logging handler
+     */
+    @ConfigItem
+    boolean enabled;
+
+    /**
+     * Hostname/IP-Address of the Logstash/Graylog Host
+     * By default it uses UDP, prepend tcp: to the hostname to switch to TCP, example: "tcp:localhost"
+     */
+    @ConfigItem(defaultValue = "localhost")
+    public String host;
+
+    /**
+     * The port
+     */
+    @ConfigItem(defaultValue = "12201")
+    public int port;
+
+    /**
+     * GELF version: 1.0 or 1.1
+     */
+    @ConfigItem(defaultValue = "1.1")
+    public String version;
+
+    /**
+     * Whether to post Stack-Trace to StackTrace field (true/false/throwable reference [0 = throwable, 1 =
+     * throwable.cause, -1 = root cause]).
+     */
+    @ConfigItem(defaultValue = "true")
+    public String extractStackTrace;
```

Comment:
So... this can be `true` or `false` or one of three integer values?  At that point maybe an `enum` is better.

Reply:
In fact it's more complicated than this, I'll update the JavaDoc you can pass any integer and it walk throught the cause tree ...

Reply:
JavaDoc updated to explain the use of throwable reference

Reply:
You've got two values (`true` and `0`) that give the same behavior, and two fundamentally different *kinds* of values.

You might instead do something like this:

```java
    @ConfigItem(defaultValue = "true")
    public boolean extractStackTrace;

    @ConfigItem(defaultValue = "0")
    public int stackTraceSearchIndex;
```

The names are just examples, but this way each property expresses one concept, and the search index can be documented to be only used when stack trace extraction is enabled.

In general we want one property to have one type.

Reply:
I agree that it is more elegant but we will no longuer have a one-to-one correspondance between Quarkus and the library.
This can be problematic:
- when the library evolves
- for people already knowing the library
- for people looking at the documentation of the library and trying to do the same with Quarkus inetgration of it

Reply:
I implements a config change as you requested in a separate commit to be able to revert it easily. If you prefere it I can squash the commit.